core/src/Kokkos_Parallel.hpp

```diff
@@ -160,24 +160,19 @@ inline void parallel_for(
         Kokkos::Impl::is_execution_policy<ExecPolicy>::value>::type* =
         nullptr) {
   uint64_t kpID = 0;
-  if (Kokkos::Profiling::profileLibraryLoaded()) {
-    Kokkos::Impl::ParallelConstructName<FunctorType,
-                                        typename ExecPolicy::work_tag>
-        name(str);
-    Kokkos::Profiling::beginParallelFor(
-        name.get(), Kokkos::Profiling::Experimental::device_id(policy.space()),
-        &kpID);
-  }
+
+  ExecPolicy inner_policy = policy;
+  Kokkos::Tools::Experimental::begin_parallel_for(inner_policy, functor, str,
```

Comment:
It's not intrinsic, but it is required (in my opinion). And if we're putting Tuning in the most important Kokkos function, and Tuning is Experimental, we're putting Experimental in that function in some sense. That's a question I'd like @crtrott to answer on, because if the answer is "this is a problem" I just need to close the PR

Reply:
This shouldn't be in the public namespace (actually that is probably also true for beginParallelFor come to speak of it - what reason do we have to expose that call to users ?? [and yes I know David that is WELL before your time ...]). Generally I don't have a problem with this use of experimental here otherwise. But again I don't see a reason to expose this new function to users - and rather want to remove the other ones from the public namespace too.

Reply:
@crtrott: `Kokkos::Tools::Impl` or `Kokkos::Tools::Experimental::Impl`? I'm in favor of the first